invenio_rdm_records/services/permissions.py

```diff
@@ -70,8 +71,15 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
     #
     # PIDs
     #
-    can_pid_reserve = can_curate
-    can_pid_delete = can_curate
+    can_pid_create = can_curate
+    can_pid_discard = can_curate  # discard an unregistered pid
+    can_pid_delete = [
+        Admin(),
+        SuperUser(),
+        SystemProcess(),
+        RecordOwnersIfExternalPID(),
```

Comment:
I would follow a but more closely the what we have for public vs restricted records:

``IfProvider(scheme='doi', provider='external', then_=can_curate)``

This will more easily allow others to reuse it I tihnk.

Similarly below:

``IfProvider(scheme='doi', provider='datacite', then_=IfPIDstatus(status=pidstatus.NEW, then_=can_curate))``

Reply:
In the end the generators are not needed.